dist/common/scripts/scylla_util.py

```diff
@@ -106,6 +110,137 @@ def curl(url, byte=False):
                 raise
 
 
+class gcp_instance:
+    """Describe several aspects of the current GCP instance"""
+
+    from psutil import virtual_memory
+    from psutil import cpu_count
+
+    def is_gce_instance(self):
+      """Check if it's GCE instance via DNS lookup to metadata server.
+      """
+      import socket
+      try:
+        addrlist = socket.getaddrinfo('metadata.google.internal', 80)
+      except socket.gaierror:
+        return False
+      for res in addrlist:
+        af, socktype, proto, canonname, sa = res
+        if (af == socket.AF_INET):
+          addr, port = sa
+          if (addr == "169.254.169.254"):
+            return True
+      return False
+
+    def __memory_size_gb(self):
+        mem = psutil.virtual_memory()
+        return mem.total/1024/1024/1024
+
+    def __cpu_count(self):
+        return psutil.cpu_count()
+
+    def __instance_metadata(self, path):
+        return curl("http://169.254.169.254/computeMetadata/v1/instance/" + path+"?recursive=true", headers="Metadata-Flavor: Google")
+        #169.254.169.254 is metadata.google.internal
+
+    def __disk_name(self, dev):
+        name = re.compile(r"(?:/dev/)?(?P<devname>[a-zA-Z]+)\d*")
+        return name.search(dev).group("devname")
+
+    def __device_exists(self, dev):
+        if dev[0:4] != "/dev":
+            dev = "/dev/%s" % dev
+        return os.path.exists(dev)
+
+    def get_file_size(self, filename):
```

Comment:
That's a little unorthodox way of finding a file size? Did you consider `fstat` instead? 
See here: https://docs.python.org/3/library/os.html#os.stat_result
And see an example of the usage in perftune.py: __learn_directory()

Reply:
doesn't work, I rewrote it to use sysfs instead